topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -229,11 +215,29 @@ public void handle(Long oldTimerId) {
                     }
                 };
                 periodic.handle(null);
-                promise.future().onComplete(start);
                 LOGGER.info("Started");
             });
     }
 
+    void startWatcher(Promise<Void> start) {
```

Comment:
Probably better to have the signature `Future<Void> startWatcher()`, or maybe `Future<Void> startWatcher(boolean firstTime)` and complete the caller's `start` future using `onComplete()` at the call site.